std/typecons.d

```diff
@@ -805,48 +800,69 @@ if (distinctFieldNames!(Specs))
          * $(UL
          *   $(LI A negative integer if the expression `v1 < v2` is true.)
          *   $(LI A positive integer if the expression `v1 > v2` is true.)
-         *   $(LI 0 if the expression `v1 == v2` is true.))
+         *   $(LI 0 if the expression `v1 == v2` is true.)
+         *   $(LI NaN in any other case.))
+         *
+         * The return type is int, if no field is a floating point type,
+         * otherwise it is the most general floating point type participating.
          */
-        int opCmp(R)(R rhs)
-        if (areCompatibleTuples!(typeof(this), R, "<"))
-        {
-            foreach (i, Unused; Types)
-            {
-                if (field[i] != rhs.field[i])
-                {
-                    return field[i] < rhs.field[i] ? -1 : 1;
-                }
-            }
-            return 0;
-        }
-
+        auto opCmp(R)(R rhs) { mixin(opCmpImpl); return 0; }
         /// ditto
-        int opCmp(R)(R rhs) const
-        if (areCompatibleTuples!(typeof(this), R, "<"))
-        {
-            foreach (i, Unused; Types)
-            {
-                if (field[i] != rhs.field[i])
+        auto opCmp(R)(R rhs) const { mixin(opCmpImpl); return 0; }
```

Comment:
Should we have a constraint here? (it used to have one)

Reply:
and what's the motivation for using a mixin? I don't see any reason

Reply:
Yes, I forgot the constraint.
Rationale for the mixin: The functions cannot be unified (e.g. with `inout`), so this was an attempt to reduce non-trivial code duplication.